src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -580,6 +601,40 @@ void Gather::exec1DCase() {
     }
 }
 
+void Gather::execCompressedEmbedding() {
+    DEBUG_LOG(getName(), " execCompressedEmbedding");
+    auto srcMemPtr = getParentEdgeAt(GATHER_DATA)->getMemoryPtr();
+    auto idxMemPtr = getParentEdgeAt(GATHER_INDICES)->getMemoryPtr();
+    const auto* psrc = reinterpret_cast<const uint8_t*>(srcMemPtr->getData());
```

Comment:
change to `getDataAs` according to https://github.com/openvinotoolkit/openvino/pull/22020